src/qibo/ui/mpldrawer.py

```diff
@@ -0,0 +1,654 @@
+    """
+
+    return _plot_quantum_circuit(
+        schedule,
+        inits,
+        plot_params,
+        labels=labels,
+        plot_labels=plot_labels,
+        schedule=True,
+        **kwargs
+    )
+
+
+def _plot_quantum_circuit(
+    gates, inits, plot_params, labels=[], plot_labels=True, schedule=False, **kwargs
+):
+    """Use Matplotlib to plot a quantum circuit.
+    gates     List of tuples for each gate in the quantum circuit.
+              (name,target,control1,control2...). Targets and controls initially
+              defined in terms of labels.
+    inits     Initialization list of gates
+    labels    List of qubit labels. optional
+
+    kwargs    Can override plot_parameters
+    """
+
+    plot_params.update(kwargs)
+    scale = plot_params["scale"]
+
+    # Create labels from gates. This will become slow if there are a lot
+    #  of gates, in which case move to an ordered dictionary
+    if not labels:
+        labels = []
+        for i, gate in _enumerate_gates(gates, schedule=schedule):
+            for label in gate[1:]:
+                if label not in labels:
+                    labels.append(label)
+
+    nq = len(labels)
+    ng = len(gates)
+
+    wire_grid = np.arange(0.0, nq * scale, scale, dtype=float)
+
+    gate_grid = np.arange(0.0, (nq if ng == 0 else ng) * scale, scale, dtype=float)
+    ax, _ = _setup_figure(
+        nq, (nq if ng == 0 else ng), gate_grid, wire_grid, plot_params
+    )
+
+    measured = None if ng == 0 else _measured_wires(gates, labels, schedule=schedule)
+    _draw_wires(ax, nq, gate_grid, wire_grid, plot_params, measured)
+
+    if plot_labels:
+        _draw_labels(ax, labels, inits, gate_grid, wire_grid, plot_params)
+
+    if ng > 0:
+        _draw_gates(
+            ax,
+            gates,
+            labels,
+            gate_grid,
+            wire_grid,
+            plot_params,
+            measured,
+            schedule=schedule,
+        )
+
+    return ax
+
+
+def _enumerate_gates(l, schedule=False):
+    "Enumerate the gates in a way that can take l as either a list of gates or a schedule"
+    if schedule:
+        for i, gates in enumerate(l):
+            for gate in gates:
+                yield i, gate
+    else:
+        for i, gate in enumerate(l):
+            yield i, gate
+
+
+def _measured_wires(l, labels, schedule=False):
+    "measured[i] = j means wire i is measured at step j"
+    measured = {}
+    for i, gate in _enumerate_gates(l, schedule=schedule):
+        name, target = gate[:2]
+        j = _get_flipped_index(target, labels)
+        if name.startswith("M"):
+            measured[j] = i
+    return measured
+
+
+def _draw_gates(
+    ax, l, labels, gate_grid, wire_grid, plot_params, measured={}, schedule=False
+):
+    for i, gate in _enumerate_gates(l, schedule=schedule):
+        _draw_target(ax, i, gate, labels, gate_grid, wire_grid, plot_params)
+        if len(gate) > 2:  # Controlled
+            _draw_controls(
```

Comment:
We could probably improve this method by adding an option to draw gates involving $\geq 2$ qubits but without forcing the procedure to draw controls. What I mean is that there are in Qibo general unitaries which we may want to draw as longer gates touching more than one qubit (rectangles instead of squares). Of course we can keep this for a future work/PR :)

ATM those general unitaries are quite confusing (in my opinion) because we draw them adding an "U" (or a square in this plotting procedure) for each qubit involved in the process.

Reply:
I think this could be a deep change on the code and we should talk about it because the unitary symbol is also comfusing when using the `print()` function, maybe the change should be to use another symbol than  a boxed "U" in my case. Maybe this is suitable for another PR as the issue  below.

If this is fine, we can close it by now...